src/Tests/Microsoft.Diagnostics.Monitoring.Profiler.UnitTests/ExceptionTrackingTests.cs

```diff
@@ -0,0 +1,196 @@
+            adapter.Environment.Add(ProfilerHelper.ClrEnvVarEnableProfiling, ProfilerHelper.ClrEnvVarEnabledValue);
+            adapter.Environment.Add(ProfilerHelper.ClrEnvVarProfiler, ProfilerHelper.Clsid.ToString("B"));
+            adapter.Environment.Add(ProfilerHelper.ClrEnvVarProfilerPath64, profilerPath);
+            adapter.Environment.Add(ProfilerHelper.ProfilerEnvVarRuntimeId, Guid.NewGuid().ToString("D"));
+            adapter.Environment.Add(ProfilerHelper.ProfilerEnvVarStdErrLoggerLevel, LogLevel.Trace.ToString("G"));
+
+            List<string> outputLines = new();
+
+            Action<string> receivedStdErrLine = (line) =>
+            {
+                if (!string.IsNullOrEmpty(line))
+                {
+                    if (line.StartsWith(ProfilerOutputPrefix, StringComparison.Ordinal))
+                    {
+                        outputLines.Add(line);
+                    }
+                    else
+                    {
+                        outputLines.Add($"{IngoreOutputPrefix}{line}");
+                    }
+                }
+            };
+
+            adapter.ReceivedStandardErrorLine += receivedStdErrLine;
+
+            using CancellationTokenSource startTokenSource = new(CommonTestTimeouts.StartProcess);
+            await adapter.StartAsync(startTokenSource.Token);
+
+            using CancellationTokenSource waitForExitSource = new(CommonTestTimeouts.WaitForExit);
+            await adapter.ReadToEnd(waitForExitSource.Token);
+
+            adapter.ReceivedStandardErrorLine -= receivedStdErrLine;
+
+            Directory.CreateDirectory("Outputs");
+            File.WriteAllLines(Path.Combine("Outputs", $"{scenarioName}.txt"), outputLines);
+
+            string[] baselineLines = File.ReadAllLines(Path.Combine("Baselines", $"{scenarioName}.txt"));
+
+            try
+            {
+                // The current index in the list of lines
+                int baselineIndex = 0;
+                int outputIndex = 0;
+
+                // The count of non-ignored lines
+                int baselineCount = 0;
+                int outputCount = 0;
+
+                // The current line value
+                string baselineLine = null;
+                string outputLine = null;
+
+                // Try to read a line from each list and compare them
+                while (TryReadNextLine(baselineLines, out baselineLine, ref baselineIndex, ref baselineCount) &&
+                       TryReadNextLine(outputLines, out outputLine, ref outputIndex, ref outputCount))
+                {
+                    try
+                    {
+                        Assert.Equal(baselineLine, outputLine);
+                    }
+                    catch (XunitException)
+                    {
+                        _outputHelper.WriteLine($"Difference on line {baselineIndex + 1} of baseline.");
```

Comment:
Should we have the `+1` here? We increment index on L161 after getting the current line, which leads me to believe if we had a conflict on the 0th index of `baselineLines`, `baselineIndex` would already be set to 1?

Reply:
Good catch! This code went through several refactorings, but this one slipped through.